src/error.c

```diff
@@ -42,6 +46,15 @@ stumpless_get_error_id( const struct stumpless_error *err ) {
   return err->id;
 }
 
+const char *
+stumpless_get_error_id_string( enum stumpless_error_id id) {
+  if ( stumpless_error_enum_to_string[id] ) {
```

Comment:
This conditional will not actually prevent a segfault if an invalid id is passed in, as it still attempts to use it to index the array. To perform a check on the id you would need to cast it to an int and make sure it is within the valid bounds, something like:
`if( id >= 0 && id < sizeof(stumpless_error_enum_to_string)/sizeof(char *) ){`